tests/sentry/incidents/endpoints/test_organization_alert_rule_details.py

```diff
@@ -389,5 +389,27 @@ def test_simple(self):
             self.get_valid_response(self.organization.slug, self.alert_rule.id, status_code=204)
 
         assert not AlertRule.objects.filter(id=self.alert_rule.id).exists()
-        assert not AlertRule.objects_with_deleted.filter(name=self.alert_rule.name)
-        assert AlertRule.objects_with_deleted.filter(id=self.alert_rule.id).exists()
+        assert not AlertRule.objects_with_archived.filter(name=self.alert_rule.name)
+        assert AlertRule.objects_with_archived.filter(id=self.alert_rule.id).exists()
+
+    def test_archive_and_create_new_with_same_name(self):
```

Comment:
I'd usually put these tests at the logic level. That way if the endpoint is removed, but we use this function elsewhere it still remains tested. Also saves you needing to duplicate the test in two places.